include/bitboard.hpp

```diff
@@ -16,6 +16,9 @@ namespace scy {
 
 class Bitboard {
    public:
+    // constructor with a default value of 0
+    constexpr Bitboard(std::uint64_t b = 0) : board(b) {}
```

Comment:
Don't do it like this.

The default, no argument constructor for `Bitboard` should just sent the value of `board` to `0`.

There should be another constructor that sets `Bitboard` to any arbitrary `std::uint64_t` value.

Also, mark these constructors as `noexcept`.

Reply:
I do not think a constructor with `std::uint64_t` is necessary anymore. I mainly added it because i wanted to test the print functionality. What we need is that default be initialised as 0, we do not need an explicit constructor for to do that. See : https://github.com/isocpp/CppCoreGuidelines/blob/master/CppCoreGuidelines.md#c45-dont-define-a-default-constructor-that-only-initializes-data-members-use-default-member-initializers-instead

Reply:
So, following the link I will remove the constructor all together. Also introduce two function `set` and `unset` which set and unset the respective positions on the board.